File: app/src/main/java/com/xomena/cmpfutboltfe/MarkerItem.java
package com.xomena.cmpfutboltfe;

import com.google.android.gms.maps.model.LatLng;
import com.google.maps.android.clustering.ClusterItem;

/**
 * CmpFutbolTfe
 * Created by andriy on 11/23/14.
 */
public class MarkerItem implements ClusterItem {
    private final LatLng mPosition;
    private String name;
    private String snippet;
    private FootballField ff;

    public MarkerItem(double lat, double lng) {
        mPosition = new LatLng(lat, lng);
    }

    @Override
    public LatLng getPosition() {
        return mPosition;
    }

    public void setName(String  n){
        this.name = n;
    }

    public void setSnippet(String s){
        this.snippet = s;
    }

    public void setFootballField(FootballField f){
        this.ff = f;
    }

    public String getName(){
        return this.name;
    }

    public String getTitle() { return this.name; }

    public String getSnippet(){
        return this.snippet;
    }

    public FootballField getFootballField(){
        return this.ff;
    }
}

File: app/src/main/java/com/xomena/cmpfutboltfe/model/MarkerItem.kt
package com.xomena.cmpfutboltfe.model

import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem

class MarkerItem(
    private val position: LatLng
) : ClusterItem {

    constructor(lat: Double, lng: Double) : this(LatLng(lat, lng))

    private var _name: String = ""
    private var _snippet: String = ""
    var footballField: FootballField? = null

    fun setName(n: String) {
        _name = n
    }

    fun setSnippet(s: String) {
        _snippet = s
    }

    fun getName(): String = _name

    override fun getPosition(): LatLng = position

    override fun getTitle(): String = _name

    override fun getSnippet(): String = _snippet
}
